.github/workflows/release.yml

```diff
@@ -0,0 +1,157 @@
+
+      - name: Stop container
+        if: always()
+        run: docker stop ${{ steps.create.outputs.containerId }}
+
+  load-target-group:
+    runs-on: ubuntu-latest
+    needs: build-group-publish
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+    - name: Checkout ${{ github.ref }}
+      uses: actions/checkout@v2
+
+      # *** comments for: target-group.json
+    - name: Load target group data
+      id: set-matrix
+      run: .github/scripts/matrix-from-json.sh .github/config/target-group.json
+
+  target-group-install:
+    runs-on: ubuntu-latest 
+    needs: load-target-group
+    strategy:
+      matrix: ${{fromJson(needs.load-target-group.outputs.matrix)}}
+
+    env:
+      AO_TOKEN_PROD: ${{ secrets.AO_TOKEN_PROD }}
+
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v2
+
+      - name: Create Container ${{ matrix.image }}
+        id: create
+        run: .github/scripts/container-from-matrix-input.sh ${{ matrix.image }}
+
+      - name: NPM Install Production
+        run: docker exec ${{ steps.create.outputs.containerId }} npm install --unsafe-perm --s3_host=production
+
+      # - name: Run tests
+      #   run: docker exec ${{ steps.create.outputs.containerId }} npm test TODO: make decision on this
+
+      - name: Stop container
+        if: always()
+        run: docker stop ${{ steps.create.outputs.containerId }}
+
+  npm-publish:
+    runs-on: ubuntu-latest 
+    if: ${{ startsWith(github.ref, 'refs/tags/') }} # will not run when manually triggered
+    needs: target-group-install
+
+    env:
+      AO_TOKEN_PROD: ${{ secrets.AO_TOKEN_PROD }}
+
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v2
+
+      - name: Create Container ${{ matrix.image }}
+        id: create
+        run: .github/scripts/container-from-matrix-input.sh node
+
+      # the file is created in the container. it should be cleared when the container is "garbage collected" by GitHub
+      - name: Set NPM Token
+        run: |
+          docker exec ${{ steps.create.outputs.containerId }} bash -c "echo '//registry.npmjs.org/:_authToken=${{ secrets.NPM_AUTH_TOKEN }}' >> .npmrc"
+          docker exec ${{ steps.create.outputs.containerId }} bash -c "echo 'registry=https://registry.npmjs.org/' >> .npmrc"
+          docker exec ${{ steps.create.outputs.containerId }} bash -c "echo 'always-auth=true '>> .npmrc"
+
+      # *** IMPORTANT: 
+      # by default our package is tagged with 'latest' to set other pass --tag. 
+      # the only pre-relese --preid triggering this workflow is alpha
+      # package is scoped to organization (@appoptics/apm-binding) set --access public to avoid 402 Payment Required
+      - name: NPM Publish (alpha)
+        run: docker exec ${{ steps.create.outputs.containerId }} npm publish --tag alpha --access public
+        if: ${{ contains(github.ref, '-') }}
```

Comment:
Not sure about that.

The goal is to make the `NPM Publish (latest)` step so that it will *never* publish a pre-release (that is a version tag that has `-` in it) with the npm `latest` tag. Right now, the filter at the top allows only `-alpha` tagged pre-release down the steps, but, this may change and it will be better to keep the negative case wider.

That said, the fact that any pre-release is npm tagged with `alpha` is confusing. 
Parsing `github.ref` is needed...

Reply:
I'm going with npm tagging that matches the conditional.
If there is a `-` in the version it is `prerelease`
If there is none it is `latest`

Commit pushed.

Note: the `prerelease` tag is only for maintainers of this package and the upstream [appoptics-apm](https://www.npmjs.com/package/appoptics-apm). There aren't really any scenarios for end user usage.

Reply:
> The goal is to make the NPM Publish (latest) step so that it will never publish a pre-release (that is a version tag that has - in it) with the npm latest tag. Right now, the filter at the top allows only -alpha tagged pre-release down the steps, but, this may change and it will be better to keep the negative case wider.

Yes, that applies to the `NPM Publish (latest)` step. But the above one says it is a `NPM Publish (alpha)` step. If we want to be future proof here, then we could change the name of the step to `NPM Publish prerelease` for the sake of consistency.